packages/editor-preview/src/browser/editor-preview-manager.ts

```diff
@@ -14,21 +14,32 @@
 // SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
 // *****************************************************************************
 
-import { EditorManager, EditorOpenerOptions, EditorWidget } from '@theia/editor/lib/browser';
-import { inject, injectable, postConstruct } from '@theia/core/shared/inversify';
+import { EditorManager, EditorOpenerOptions, EditorWidget, Range } from '@theia/editor/lib/browser';
+import { inject, injectable, named, postConstruct } from '@theia/core/shared/inversify';
 import { EditorPreviewPreferences } from './editor-preview-preferences';
-import { MaybePromise } from '@theia/core/lib/common';
+import { ContributionProvider, MaybePromise, Prioritizeable, RecursivePartial } from '@theia/core/lib/common';
 import URI from '@theia/core/lib/common/uri';
 import { EditorPreviewWidgetFactory, EditorPreviewOptions } from './editor-preview-widget-factory';
 import { EditorPreviewWidget } from './editor-preview-widget';
 import { FrontendApplicationStateService } from '@theia/core/lib/browser/frontend-application-state';
 
+export const EditorSelectionResolver = Symbol('EditorSelectionResolver');
+export interface EditorSelectionResolver {
```

Comment:
We should add a TS doc what a `EditorSelectionResolver` actually is and that it only applies if there is not yet a selection.